src/main/scala/sigmastate/utils/ByteReader.scala

```diff
@@ -172,6 +172,10 @@ class ByteBufferReader(buf: ByteBuffer) extends ByteReader {
   @inline override def position_=(p: Int): Unit = buf.position(p)
 
   @inline override def remaining: Int = buf.remaining()
+
+  private val lvl: AtomicInteger = new AtomicInteger(0)
```

Comment:
@greenhat Our Reader/Writer implementations are not thread safe, we should probably mention this in ScalaDoc. Because of this making `level` safe doesn't make much sense IMO. 
I also cannot make up a practical case for two threads working with the same Reader instance.

Reply:
Agree. That was an overkill.